includes/multi-currency/MultiCurrency.php

```diff
@@ -845,6 +845,54 @@ public function get_price( $price, string $type ): float {
 		return $this->get_adjusted_price( $converted_price, $apply_charm_pricing, $currency );
 	}
 
+	/**
+	 * Gets a raw converted amount based on the amount and currency codes passed.
+	 * This is a helper method for external conversions, if needed.
+	 *
+	 * @param float  $amount        The amount to be converted.
+	 * @param string $to_currency   The 3 letter currency code to convert the amount to.
+	 * @param string $from_currency The 3 letter currency code to convert the amount from.
+	 *
+	 * @return float The converted amount.
+	 *
+	 * @throws InvalidCurrencyException
+	 * @throws InvalidCurrencyRateException
+	 */
+	public function get_raw_conversion( float $amount, string $to_currency, string $from_currency = '' ): float {
+		$enabled_currencies = $this->get_enabled_currencies();
+
+		// If the from_currency is not set, use the store currency.
+		if ( '' === $from_currency ) {
+			$from_currency = $this->get_default_currency()->get_code();
+		}
+
+		// We throw an exception if either of the currencies are not enabled.
+		$to_currency   = strtoupper( $to_currency );
+		$from_currency = strtoupper( $from_currency );
+		foreach ( [ $to_currency, $from_currency ] as $code ) {
+			if ( ! isset( $enabled_currencies[ $code ] ) ) {
+				$message = 'Invalid currency passed to get_raw_conversion: ' . $code;
+				Logger::error( $message );
+				throw new InvalidCurrencyException( $message, 'wcpay_multi_currency_invalid_currency', 500 );
+			}
+		}
+
+		// Get the rates.
+		$to_currency_rate   = $enabled_currencies[ $to_currency ]->get_rate();
+		$from_currency_rate = $enabled_currencies[ $from_currency ]->get_rate();
+
+		// Thrown an exception in case from_currency_rate is less than or equal to 0.
+		if ( 0 >= $from_currency_rate ) {
```

Comment:
Noting that `0 === $from_currency_rate` does not work here since `$from_currency_rate` is a float. I also figured it would be better to make sure we weren't doing any sort of negative conversion.

Reply:
Makes sense, `get_rate` will always return a float, so no need for checking the other types. Thanks for also adding the negatives to the check!